### PR TITLE
Visual parsing

### PR DESCRIPTION
**Pull request by lukedirtwalker**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/158%23discussion_r40601547%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/158%23issuecomment-143869682%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/158%23issuecomment-143870837%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/158%23issuecomment-143869682%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22OK%2C%20it%27s%20clearly%20a%20work%20in%20progress%2C%20but%20I%27ll%20merge%20so%20that%20I%20can%20also%20fix%20things.%20Just%20before%20I%20do%20it%2C%20could%20you%20please%20remove%20the%20optnone%20attribute%20that%20you%20flagged%3F%20I%20don%27t%20think%20this%20will%20compile%20in%20gcc.%22%2C%20%22created_at%22%3A%20%222015-09-28T20%3A49%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Removed%20optnone%20flag%22%2C%20%22created_at%22%3A%20%222015-09-28T20%3A54%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2086f8103e320023a5962539d0ef0e4e74d4ede848%20InformationScripting/src/handlers/HQuery.cpp%2097%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/158%23discussion_r40601547%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Remove%20%22%2C%20%22created_at%22%3A%20%222015-09-28T20%3A31%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20InformationScripting/src/handlers/HQuery.cpp%3AL1-124%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 86f8103e320023a5962539d0ef0e4e74d4ede848 InformationScripting/src/handlers/HQuery.cpp 97'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/158#discussion_r40601547'>File: InformationScripting/src/handlers/HQuery.cpp:L1-124</a></b>
- <a href='https://github.com/lukedirtwalker'></a> Remove
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/158#issuecomment-143869682'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'></a> OK, it's clearly a work in progress, but I'll merge so that I can also fix things. Just before I do it, could you please remove the optnone attribute that you flagged? I don't think this will compile in gcc.
- <a href='https://github.com/lukedirtwalker'></a> Removed optnone flag

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/158?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/158?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/158'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/158'>Refresh</a>.<a href='#crh-update-none'></a>

File: InformationScripting/src/handlers/HQuery.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+QueryNodeContainer* HQuery::parentContainer(InformationScripting::QueryNode* e)
+{
+	// Is this expression part of an expression statement
+	auto ep = e->parent();
+	while (ep && !DCast<QueryNodeContainer>(ep)) ep = ep->parent();
+
+	return DCast<QueryNodeContainer>(ep);
+}
+
+__attribute__((optnone))
```
Comment (lukedirtwalker) [inline InformationScripting/src/handlers/HQuery.cpp]: Remove

**Comment by dimitar-asenov** [comment]
OK, it's clearly a work in progress, but I'll merge so that I can also fix things. Just before I do it, could you please remove the optnone attribute that you flagged? I don't think this will compile in gcc.

**Comment by lukedirtwalker** [comment]
Removed optnone flag

**Pull request merged**